### PR TITLE
remove references to 'source repository'

### PR DESCRIPTION
**Pull request by MHBauer**
- rewrite intro to Dockerfile reference usage section to remove
  references to 'source repository'
- docs for #8648
  Signed-off-by: Morgan Bauer mbauer@us.ibm.com

### CODE REVIEW

**Comment by thaJeztah** [comment]
Thanks for contributing, @MHBauer !

Some initial thoughts while looking through the changes

- We should avoid using `PATH` in this description, because `PATH` has a special meaning, so will lead to confusion.
- Good call on mentioning `-f`; I think that, with the right wording, we can avoid using a "Note:" for that - we try to avoid adding too many "Notes"

I think we can reduce the amount of changes needed in this document if we used "build context" more consistently; that's the term used when sending files to the daemon;

```
Sending build context to Docker daemon  2.56 kB
Sending build context to Docker daemon 
Step 0 : FROM busybox
 ---> e72ac664f4f0
....

I see some other issues with the text, but I defer to @moxiegirl for that (otherwise I'll mess it up :-))

**Comment by thaJeztah** [comment]
ping @moxiegirl if you have some time to look at this one (see my previous comment) - I know you're busy

**Comment by moxiegirl** [comment]
@MHBauer thank you for the contribution. @thaJeztah Is correct this needs a bit of tweaking.  I've got plans for my Saturday but I'll be online this afternoon and I'll put something through for sure.

**Comment by moxiegirl** [comment]
@MHBauer @thaJeztah My tool is showing those as 404 because of the hash.  :-P Dammit now I have to find a more reliable tool.

File: docs/reference/builder.md [hunk 1]
```diff
@@ -25,31 +25,36 @@ Dockerfile knowledge with the [Dockerfile tutorial](/userguide/level1).
 
 ## Usage
 
-To [*build*](/reference/commandline/cli/#build) an image from a source repository,
-create a description file called `Dockerfile` at the root of your repository.
-This file will describe the steps to assemble the image.
+The `Dockerfile` is used by
```
Comment (moxiegirl) [inline docs/reference/builder.md]: @MHBauer Thank you for the contribution.  This page is needs a few updates so glad you picked it to work on.  Much of the language can be tightened.  It has a lot of notes and warnings indicating structural issues.  

---> https://gist.github.com/moxiegirl/8f1fe9938ce2f85c984f

# Dockerfile reference

Docker can build images automatically by reading the instructions from a
`Dockerfile`. A `Dockerfile` is a text document that contains all the commands a
user could call on the command line to assemble an image. Using `docker build`
users can create an automated build that executes several command-line
instructions in successesion.

This page describes the commands you can use in a `Dockerfile`.  When you are
done reading this page, refer to the [`Dockerfile` Best
Practices](/articles/dockerfile_best-practices) for a tip-oriented guide.

## Usage

The [`docker build`](/reference/commandline/cli/#build) command builds an image from a `Dockerfile` and a *context*. The build's context is the files at a specified location `PATH` or `URL`.  The `PATH` is a directory on your local filesystem.  The `URL` is a the location of a Git repository.

A context is processed recursively. So, a `PATH` includes any subdirectories and the `URL` includes the repository and its submodules.  A simple build command that uses the current directory as context:

    $ docker build .
    Sending build context to Docker daemon  6.51 MB
    Sending build context to Docker daemon 
    ...

The build is run by the Docker daemon, not by the CLI. The first thing a build process does is send the entire context (recursively) to the daemon.  In most cases, it's best to start with empty directory as context and keep
your Dockerfile in that directory. Add only the files needed for
building the Dockerfile.

>**Warning**: Do not use your root directory, `/`, as the `PATH` as it causes
>the build to transfer the entire contents of your hard drive to the Docker
>daemon. 

To use a file in the build context, the `Dockerfile` can refer to it with
command, for example,  a `COPY` instruction. To increase the build's
performance, exclude files and directories by adding a `.dockerignore` file to
the context directory.  For information about how to [create a `.dockerignore`
file](#dockerignore-file) on this page.

Traditionally, the `Dockerfile` is called `Dockerfile` and located in the root of the context. This isn't a requirement though.  You use the `-f` flag with `docker build` to point to a file anywhere in your file sytem.

You can specify a repository and tag at which to save the new image if
the build succeeds:

**Comment by cpuguy83** [comment]
Any update on this? @moxiegirl Can we carry this?

**Comment by moxiegirl** [comment]
@cpuguy83 if you like the changes I added, I'm happy to carry.  You good with it?

**Comment by cpuguy83** [comment]
@moxiegirl LGTM

**Comment by MHBauer** [comment]
my bad, I dropped the ball in following up on this. It seems like you have the changes you want to make ready. Should I close?

**Comment by moxiegirl** [comment]
@MHBauer no worries. I'll close this with my carry.

**Pull request closed without merging**